edx_api/certificates/__init__.py

```diff
@@ -67,9 +66,6 @@ def get_student_certificates(self, username, course_ids=None):
 
         all_certificates = []
         for course_id in course_ids:
-            try:
-                all_certificates.append(self.get_student_certificate(username, course_id))
-            except HTTPError:
-                pass
+            all_certificates.append(self.get_student_certificate(username, course_id))
```

Comment:
this solution does not really work: `get_student_certificate` raises for erroneous statuses, 404 included. Now 404 is a valid status to get with the logic of `get_student_certificates` (meaning querying for certificates for ALL the enrollments).

I think that we need to intercept the exception and check the HTTP status code and raise only in case of >=500